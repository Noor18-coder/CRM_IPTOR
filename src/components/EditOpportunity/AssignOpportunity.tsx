import React from 'react';
import { Dispatch } from 'redux';
import { useSelector, useDispatch } from 'react-redux';
import { pick } from 'lodash';
import { AppState } from '../../store/store';

import * as models from '../../helpers/Api/models';
import UserSearchField from '../Shared/Search/UserSearchField';
import { editOpportunity } from '../../store/OpportunityDetails/Actions';

const AssignOpportunity: React.FC = () => {
  const state: AppState = useSelector((appState: AppState) => appState);
  const dispatch: Dispatch<any> = useDispatch();
  const opportunityDetails: models.AddOpportunityDefaultParams = pick(state.opportuntyDetails.opportunityDefaultParams, ['opportunityId', 'userId']);
  const [opportunity, setOpportunity] = React.useState<models.AddOpportunityDefaultParams>(opportunityDetails);
  const [handler, setHandler] = React.useState<string | undefined>(opportunityDetails?.userId);

  const onNextButtonClick = () => {
    dispatch(editOpportunity(opportunity));
  };

  const onHandlerChange = (user: models.UserItem) => {
    if (user) {
      setOpportunity({
        ...opportunity,
        userId: user.user,
      });
      setHandler(handler);
    } else {
      setOpportunity({
        ...opportunity,
        userId: '',
      });
    }
  };

  return (
    <>
      <div className="assign-opportunity-form">
        <div className="form-group oppty-form-elements">
          <label htmlFor="owner" className="opp-label">
            Assign Opportunity
          </label>
          <UserSearchField onChange={onHandlerChange} description="Owner" />
        </div>
      </div>

      <div className="step-nextbtn-with-arrow stepsone-nxtbtn">
        <button
          type="button"
          disabled={!(handler !== opportunity.userId && opportunity.userId !== '')}
          className={handler !== opportunity.userId && opportunity.userId !== '' ? 'stepone-next-btn done' : 'stepone-next-btn inactive'}
          onClick={onNextButtonClick}>
          Done
        </button>
      </div>
    </>
  );
};

export default AssignOpportunity;
