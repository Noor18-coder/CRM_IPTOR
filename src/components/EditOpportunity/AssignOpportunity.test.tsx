import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import AssignOpportunity from './AssignOpportunity';
import StoreMock from '../../mocks/Store.mock';

const store = StoreMock.createInitialStore();

describe('[EditOpportunity] AssignOpportunity', () => {
  it('should renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <AssignOpportunity />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
