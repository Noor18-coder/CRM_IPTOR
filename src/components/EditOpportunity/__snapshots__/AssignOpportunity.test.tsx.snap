// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[EditOpportunity] AssignOpportunity should renders correctly 1`] = `
Array [
  <div
    className="assign-opportunity-form"
  >
    <div
      className="form-group oppty-form-elements"
    >
      <label
        className="opp-label"
        htmlFor="owner"
      >
        Assign Opportunity
      </label>
      <input
        className="form-control"
        onFocus={[Function]}
        type="text"
        value=""
      />
    </div>
  </div>,
  <div
    className="step-nextbtn-with-arrow stepsone-nxtbtn"
  >
    <button
      className="stepone-next-btn inactive"
      disabled={true}
      onClick={[Function]}
      type="button"
    >
      Done
    </button>
  </div>,
]
`;
